pageserver/src/tenant.rs

```diff
@@ -431,11 +479,29 @@ impl Tenant {
             // missing locally, and scheduling uploads for anything that's missing
             // in remote storage.
             timeline
-                .reconcile_with_remote(metadata, index_part, first_save)
+                .reconcile_with_remote(
+                    up_to_date_metadata,
+                    remote_startup_data.as_ref().map(|r| &r.index_part),
+                )
                 .await
                 .context("failed to reconcile with remote")?
         }
 
+        // Save the metadata file to local disk.
+        // Do this last, so that if we crash in-between, we won't think that the
+        // local state is valid.
+        // FIXME, we wont anyway since we have attach mark file. Can still do this at the end
```

Comment:
Should we change this, or leave as is? Any objections to leave as is?

Reply:
What would be the alternative? Update the metadata file earlier already? 

Seems fine as it is.